kong/resty/dns/client.lua

```diff
@@ -839,7 +814,37 @@ local function syncQuery(qname, r_opts, try_list)
   local key = qname..":"..r_opts.qtype
   local item = queue[key]
 
-  -- if nothing is in progress, we start a new async query
+  local is_yieldable = in_yieldable_phase()
+
+  -- If nothing is in progress and in the yieldable phase,
+  -- we start a new sync query
+
+  if not item and is_yieldable then
+    item = {
+      key = key,
+      semaphore = semaphore(),
+      qname = qname,
+      r_opts = cycle_aware_deep_copy(r_opts),
+      try_list = try_list,
+    }
+    queue[key] = item
```

Comment:
I presume this is also done in `asyncQuery()`. In that case, we'd better extract the part that handles queue and semaphore and not copy&paste.

Reply:
A possible name for the extracted function: "`acquire_lock_query`" and "`release_lock_query`".

Reply:
i agree, but refactoring the original logic is not the goal of this PR. And the task of improving this library is in the plan.